contracts/utils/helpers.cairo

```diff
@@ -0,0 +1,373 @@
+        bitwise_ptr : BitwiseBuiltin*,
+        pedersen_ptr : HashBuiltin*,
+        range_check_ptr
+    }(
+        game : felt
+    ):
+    alloc_locals
+    let (local game_info) = stored_game.read(
+        game_id=game, generation=0)
+    let game_id = game_info[1]
+
+    with_attr error_message(
+        "Game with genesis state {game} already exists"
+    ):
+        assert_not_equal(game_id, game)
+    end
+
+    let (high, _) = split_felt(game)
+    let (invalid_bits, _) = unsigned_div_rem(high, 2**97)
+    with_attr error_message(
+        "Game {game} is not a valid game"
+    ):
+        assert invalid_bits = 0
+    end
+    return ()
+end
+
+func get_last_state{
+        syscall_ptr : felt*,
+        pedersen_ptr : HashBuiltin*,
+        range_check_ptr
+    }() -> (
+        current_generation : felt,
+        current_state : felt
+    ):
+    let (generation) = current_generation.read(
+        game_id=INFINITE_GAME_GENESIS
+    )
+    let (game_info) = stored_game.read(
+        game_id=INFINITE_GAME_GENESIS,
+        generation=generation
+    )
+    return(generation, game_info[1])
+end
+
+func assert_valid_cell_index{
+        syscall_ptr : felt*,
+        pedersen_ptr : HashBuiltin*,
+        range_check_ptr
+    }(
+        cell_index : felt
+    ):
+    with_attr error_message(
+        "Cell index {cell_index} out of range"
+    ):
+        assert_le_felt(cell_index, DIM*DIM-1)
+    end
+
+    return ()
+end
+
+func create_new_game{
+        syscall_ptr : felt*,
+        pedersen_ptr : HashBuiltin*,
+        range_check_ptr
+    }(
+        game_state : felt,
+        user_id : felt
+    ):
+    save_game(
+        game_id=game_state,
+        generation=0,
+        user_id=user_id,
+        packed_game=game_state
+    )
+    save_generation(
+        game_id=game_state,
+        generation=0
+    )
+    game_created.emit(
+        owner_id=user_id,
+        game_id=game_state,
+        state=game_state
+    )
+    return ()
+end
+
+# User input may override state to make a cell alive.
+func activate_cell{
+        syscall_ptr : felt*,
+        bitwise_ptr : BitwiseBuiltin*,
+        pedersen_ptr : HashBuiltin*,
+        range_check_ptr
+    }(
+        generation : felt,
+        caller : felt,
+        cell_index : felt,
+        current_state : felt
+    ):
+    alloc_locals
+
+    assert_valid_cell_index(cell_index)
+    let (packed_game) = pack_single_cell(
+        cell_index=cell_index,
+        current_state=current_state
+    )
+
+    with_attr error_message(
+        "No changes made to the game"
+    ):
+        assert_not_equal(current_state, packed_game)
+    end
+
+    save_game(
+        game_id=INFINITE_GAME_GENESIS,
+        generation=generation,
+        user_id=caller,
+        packed_game=packed_game
+    )
+
+    cell_revived.emit(
+        user_id=caller,
+        generation=generation,
+        cell_index=cell_index,
+        state=packed_game
+    )
+
+    return ()
+end
+
+func evolve_and_save{
```

Comment:
Using and in function name is often a clue it should be split into two. This is the case here as well.